frontend/views/pages/Payments.vue

```diff
@@ -141,22 +157,38 @@ export default {
       return this.ourGroupProfile.incomeDetailsType === 'incomeAmount'
     },
     tabItems () {
-      const items = [{
-        title: L('Todo'),
-        url: 'PaymentRowTodo',
-        notification: this.paymentsTodo.length
-      }, {
-        title: L('Sent'),
-        url: 'PaymentRowSent',
-        notification: this.paymentsSent.length
-      }]
-      if (this.paymentsReceived.length > 0) {
+      const items = []
+
+      if (!this.needsIncome) {
+        items.push({
+          title: L('Todo'),
+          url: 'PaymentRowTodo',
+          notification: this.paymentsTodo.length
+        })
+
+        items.push({
+          title: L('Sent'),
+          url: 'PaymentRowSent'
+        })
+      }
+
+      const doesNotNeedIncomeAndDidReceiveBefore = !this.needsIncome && this.paymentsReceived.length
+      const doesNeedIncomeAndDidSentBefore = this.needsIncome && this.paymentsSent.length
+
+      if (doesNotNeedIncomeAndDidReceiveBefore || doesNeedIncomeAndDidSentBefore) {
         items.push({
           title: L('Received'),
-          url: 'PaymentRowReceived',
-          notification: this.paymentsReceived.length
+          url: 'PaymentRowReceived'
+        })
+      }
+
+      if (doesNeedIncomeAndDidSentBefore) {
+        items.push({
```

Comment:
This is created twice, here and on line 169, which are mutually exclusive but are in the same scope. This should probably be cleaned up if possible. I think removing the two variables on lines 175 and 176 would help, by making it possible to combine the conditions above and below it, which would also make semantics more clear.